pkg/server/tests/cursor/cursor_test.go

```diff
@@ -226,3 +226,66 @@ outerLoop:
 		}
 	}
 }
+
+func TestSerialLazyExecuteAndFetch(t *testing.T) {
```

Comment:
Now we only have a serial test here. This PR won't test concurrent cursor fetch with normal execution (or another cursor fetch). After both https://github.com/pingcap/tidb/pull/54456 and this one are merged, I'll add more tests about the concurrent lazy cursor fetch.